### ISSUE TITLE
Cannot install released version 2.1.0.1

### ISSUE THREAD

**Issue by ycecube** [docker/for-win#4480]
Hi,

I've tried to update from a previous version first to 2.1.0.0 and now 2.1.0.1, but both failed to me.

Now it throws the following error:

```
The system cannot find the file specified. (Exception from HRESULT: 0x80070002)
   at System.Runtime.InteropServices.ComTypes.IPersistFile.Save(String pszFileName, Boolean fRemember)
   at CommunityInstaller.ShortcutAction.DoAsync(CancellationToken cancelToken)
   at CommunityInstaller.InstallWorkflow.<HandleD4WPackage>d__26.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at CommunityInstaller.InstallWorkflow.<ProcessAsync>d__23.MoveNext()
```

Windows version: 10.0.18362.267 (1903)
WSL Ubuntu version: Linux DESKTOP-9I8GEF0 4.4.0-18362-Microsoft #1-Microsoft Mon Mar 18 12:02:00 PST 2019 x86_64 x86_64 x86_64 GNU/Linux

**Comment by PSC77** [comment]
Same issue here during a fresh install.

**Comment by whereisaaron** [comment]
I installed the update and poof, error display and docker no long working or installed. Tried installing fresh, might be working, but system tray menu won't appear anymore 😢

**Comment by ycecube** [comment]
> For me disabling Windows Defender worked (i.e. allowing docker to access certain system folders)...

That doesn't sound too great, I would rather not allow docker poking in system folders if not necessary, even so if defender disallows it.

**Comment by mat007** [comment]
Thanks for the report, it seems that Windows Defender is preventing the installer from creating the Start Menu shortcut on some machines…
@PSC77 can you confirm you saw this same error before disabling Windows Defender and making it work?

**Comment by mat007** [comment]
It seems this may be due to the Ransomware protection of Windows Defender, apparently if you turn this off during the installation of Docker Desktop it will complete successfully. You can turn it back on afterwards.

![image](https://user-images.githubusercontent.com/2386884/62822513-fc087280-bb84-11e9-9ac4-1e6336e3241c.png)

Apparently all our CI machines have this turned off, I’ll try and see how to test this properly and if we can work around this in our installer.

**Comment by ralusnom** [comment]
I had exactly the same problem. Disabling the Windows Defender completely let me install Docker.

Also, I could not disable Windows Defender through the settings because I have a managed domain account and therefore I'm not authorized. I could deactivate it however over the Registry. Bad boy 👿 

- WIN+R, open `regedit` 
- Goto `HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\Windows Defender`
- Create a `DWORD (32-bit) Value` key with the name **DisableAntiSpyware** and value 1. 
- Restart system
- Install Docker

Of course, you should remove the key later and not tell the domain administrator what you did.

**Comment by jmartens** [comment]
Possibly a duplicate of #4472?

**Comment by mat007** [comment]
I don’t think so @jmartens, #4472 happens when uninstalling the previous version, this one is when installing the new one.

**Comment by tamasd** [comment]
I have the real time scanning (and with it the ransomware protection) disabled for other reasons, and I never experienced this issue. So you don't have to disable the whole defender, only the real time scanning part (which can be disabled in group policy).

**Comment by juliusstoerrle** [comment]
Just for further information: ![grafik](https://user-images.githubusercontent.com/25788444/62829910-6f04fe00-bc04-11e9-88fc-aa7e1ba879e4.png)

Unblocking the action and restarting the installation process results in the installation beeing succesful

**Comment by ycecube** [comment]
Looks like it is also enough if you just untick the "Add shortcut to Desktop" option, this solved the installation issue for me.

By the way, why everyone thinks that it is okay to disable a security setting on your OS just to be able to install something? Okay, I mean I trust docker... but in general.

**Comment by talib570** [comment]
For me disabling `Controlled Folder Access` worked

**Comment by whereisaaron** [comment]
For me completely uninstalling 2.1.0.1, then installing it again got the system tray menu to work again.

**Comment by miltlath** [comment]
i follow yours instruction to switch off the Ransomware protection of Windows Defender and finish the installation successful . thank you all of us so much.

**Comment by mehrabanb** [comment]
after re-enabling ransomware protection it might still block docker , so you need to goto -> Windows Security -> Virus and Threat protection -> Ransomware protection -> Manage ransomeware protection -> block history -> click on allow (needs admin access)

**Comment by paj777** [comment]
I have attempted to install 2.1.0.5 and I am getting the same error.  I have completely disabled windows security via the registry and this has not fixed the issue.  Same error as above, Windows 1909.
